src/Features/Core/SolutionCrawler/WorkCoordinator.HighPriorityProcessor.cs

```diff
@@ -114,7 +114,7 @@ protected override Task WaitAsync(CancellationToken cancellationToken)
                             // see whether we have work item for the document
                             WorkItem workItem;
                             CancellationTokenSource documentCancellation;
-                            Contract.ThrowIfFalse(GetNextWorkItem(out workItem, out documentCancellation));
+                            Contract.ThrowIfFalse(TryGetNextWorkItem(out workItem, out documentCancellation));
```

Comment:
I'm not sure I understand this. Is it OK *in general* for `TryGetNextWorkItem` to return `false`, *in this specific case* we know it won't?